tests/snp_launch.rs

```diff
@@ -3,138 +3,34 @@
-
-    #
-    # We shouldn't come back from the VMGEXIT, but if we do, just loop.
-    #
-3:
-    hlt
-    jmp     3b
-4:
-.fill(({CODE_SIZE} - (4b - 2b)))
-    ",
-    CODE_SIZE = const CODE_SIZE,
-    options(noreturn)
-    )
-}
- */
+// has to be a multiple of 16
+const CODE: &[u8; 16] = &[
+    0xf4; 16 // hlt
+];
 
 #[cfg_attr(not(has_sev), ignore)]
 #[test]
 fn snp() {
-    fn set_cpu(kvm_fd: &Kvm, vcpu_fd: &mut VcpuFd) {
-        vcpu_fd
-            .set_cpuid2(&kvm_fd.get_supported_cpuid(KVM_MAX_CPUID_ENTRIES).unwrap())
-            .unwrap();
-
-        let mut regs = vcpu_fd.get_regs().unwrap();
-        regs.rip = 0xFFFF_F000;
-        vcpu_fd.set_regs(&regs).unwrap();
-
-        let mut sregs = vcpu_fd.get_sregs().unwrap();
-
-        sregs.cs = KvmSegment {
-            base: 0x0,
-            limit: 0xffffffff,
-            selector: 0x8,
-            type_: 0xb,
-            present: 0x1,
-            dpl: 0x0,
-            db: 0x1,
-            s: 0x1,
-            l: 0x0,
-            g: 0x1,
-            avl: 0x0,
-            unusable: 0x0,
-            padding: 0x0,
-        };
-
-        sregs.ds = KvmSegment {
-            base: 0x0,
-            limit: 0xffffffff,
-            selector: 0x10,
-            type_: 0x3,
-            present: 0x1,
-            dpl: 0x0,
-            db: 0x1,
-            s: 0x1,
-            l: 0x0,
-            g: 0x1,
-            avl: 0x0,
-            unusable: 0x0,
-            padding: 0x0,
-        };
-
-        sregs.cr0 |= 1; // Cr0Flags::PROTECTED_MODE_ENABLE.bits();
-
-        vcpu_fd.set_sregs(&sregs).unwrap();
-    }
-
     let kvm_fd = Kvm::new().unwrap();
     let vm_fd = kvm_fd.create_vm().unwrap();
-    vm_fd.create_irq_chip().unwrap();
 
-    let mut address_space = Map::map(CODE_SIZE)
+    const MEM_SIZE: usize = 0x1000;
+    let mut address_space = Map::map(MEM_SIZE)
         .anywhere()
         .anonymously()
         .known::<perms::ReadWrite>(Kind::Private)
         .unwrap();
 
-    /*
-    let code = unsafe { std::slice::from_raw_parts(vm_code as *const u8, CODE_SIZE) };
-
-    000000000000c5a0 <snp_launch::code>:
-        c5a0:   b8 00 11 02 00          mov    eax,0x21100
-        c5a5:   31 d2                   xor    edx,edx
-        c5a7:   b9 30 01 01 c0          mov    ecx,0xc0010130
-        c5ac:   0f 30                   wrmsr
-        c5ae:   f3 0f 01 d9             vmgexit
-        c5b2:   f4                      hlt
-        c5b3:   eb fd                   jmp    c5b2 <snp_launch::code+0x12>
-        ...
-      */
-
-    let code: [u8; 21] = [
-        0xb8, 0x00, 0x11, 0x02, 0x00, 0x31, 0xd2, 0xb9, 0x30, 0x01, 0x01, 0xc0, 0x0f, 0x30, 0xf3,
-        0x0f, 0x01, 0xd9, 0xf4, 0xeb, 0xfd,
-    ];
-
-    address_space[..code.len()].copy_from_slice(&code);
+    address_space[..CODE.len()].copy_from_slice(&CODE[..]);
 
     let mem_region = kvm_userspace_memory_region {
         slot: 0,
-        guest_phys_addr: 0xFFFF_F000,
+        guest_phys_addr: 0,
```

Comment:
The address should be a constant. Also, it should probably be something other than zero.